visualization/app/codeCharta/util/nodeDecorator.ts

```diff
@@ -1,32 +1,51 @@
 "use strict"
 import * as d3 from "d3"
-import { HierarchyNode } from "d3"
+import { HierarchyNode, hierarchy } from "d3"
 import {
 	BlacklistItem,
-	BlacklistType,
 	CCFile,
 	CodeMapNode,
 	MetricData,
-	FileMeta,
 	EdgeMetricCount,
 	KeyValuePair,
 	AttributeTypes,
-	AttributeTypeValue
+	AttributeTypeValue,
+	BlacklistType
 } from "../codeCharta.model"
-import { CodeMapHelper } from "./codeMapHelper"
 import { MetricService } from "../state/metric.service"
+import { CodeMapHelper } from "./codeMapHelper"
+import ignore from "ignore"
 
 export class NodeDecorator {
-	public static decorateMap(map: CodeMapNode, fileMeta: FileMeta, metricData: MetricData[]) {
-		this.decorateMapWithMissingObjects(map, fileMeta)
+	public static decorateMap(map: CodeMapNode, metricData: MetricData[], blacklist: BlacklistItem[]) {
+		this.decorateMapWithMissingObjects(map)
 		this.decorateMapWithCompactMiddlePackages(map)
 		this.decorateLeavesWithMissingMetrics(map, metricData)
+		this.decorateMapWithBlacklist(map, blacklist)
 	}
 
 	public static preDecorateFile(file: CCFile) {
 		this.decorateMapWithPathAttribute(file)
 	}
 
+	private static decorateMapWithBlacklist(map: CodeMapNode, blacklist: BlacklistItem[]) {
+		const flattened = ignore()
+		const excluded = ignore()
+
+		for (let item of blacklist) {
+			const path = CodeMapHelper.transformPath(item.path)
+			item.type === BlacklistType.flatten ? flattened.add(path) : excluded.add(path)
+		}
+
+		hierarchy(map)
+			.descendants()
```

Comment:
@NearW I had to change this from `leaves()` to `descendants()` as the TreeView needs to know if a folder is excluded (to not show it) or is flattened (to show it in gray and the flattened symbol next to it)

Reply:
Agreed.